main/src/library/Set.flix

```diff
@@ -229,7 +229,8 @@ mod Set {
                 if (useParallelEvaluation(s))
                     let h = (kl, _, kr, _) -> g(kl, kr);
                     let Set(t) = s;
-                    let o = RedBlackTree.parMinimumBy(threads() - 1, h, t);
+                    let threads = unchecked_cast(Environment.handle(threads)() as _ \ {});
+                    let o = RedBlackTree.parMinimumBy(threads - 1, h, t);
```

Comment:
I think we should move the threads argument into RedBlackTree, so here we should just call parMinimumBy